sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if n_bins < 2.
+
+    categorical_features : int array-like (default=None)
+        Column indexes of categorical features. Categorical features
+        will not be discretized, and will remain as is after the transform.
+
+    Attributes
+    ----------
+    mins_ : float array, shape (n_features_,)
```

Comment:
I suppose `mins_` and `maxes_` can indeed be part of `bin_edges_`. If you're feeling strongly about including `mins_` and `maxes_` within `bin_edges`, I could certainly change this.